src/bsp/crypto/iotc_bsp_crypto_test_sha256.cc

```diff
@@ -0,0 +1,108 @@
+/* Copyright 2018-2019 Google LLC
+ *
+ * This is part of the Google Cloud IoT Device SDK for Embedded C,
+ * it is licensed under the BSD 3-Clause license; you may not use this file
+ * except in compliance with the License.
+ *
+ * You may obtain a copy of the License at:
+ *  https://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+
+#include "gmock.h"
+#include "gtest.h"
+#include "iotc_bsp_crypto.h"
+
+namespace iotctest {
+namespace {
+
+using ::testing::ElementsAreArray;
+
+constexpr unsigned int kSHA256HashSize = 32;
+
+TEST(IotcBspCryptoSha256, OutputBufferIsNull) {
+  EXPECT_EQ(iotc_bsp_sha256(nullptr,
+                            reinterpret_cast<const unsigned char*>("abc"),
+                            /* src_buf_size = */ 3),
```

Comment:
Condense the parameter comment a little bit:

`/*src_buf_size=*/`

Apply everywhere.